File: Website_Phim/WebPhim-Backend/MovieWebsite/MovieWebsite/src/main/java/com/project/MovieWebsite/dtos/EpisodeDTO.java

package com.project.MovieWebsite.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor

public class EpisodeDTO {

    @JsonProperty("movie_id")
    private int movieId;

    @JsonProperty("episode")
    private int episode = 1;

    @JsonProperty("movie_url")
    @NotBlank(message = "Movie url required")
    private String movieUrl;


}

File: MovieWebsiteJA-tin/MovieWebsiteJA-tin/MovieWebsite-fontend/src/app/dtos/user/episode.dto.ts
import {
    IsString,
    IsNumber,
} from 'class-validator';

export class EpisodeDTO {
    
    @IsNumber()
    movie_id: number;

    @IsNumber()
    episode: number;


    @IsString()
    movie_url: string;

    constructor(data:any) {
        this.movie_id = data.movie_id;
        this.episode =  data.episode;
        this.movie_url= data.movie_url
    }
}